src/internal/staking.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/staking}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * faucet namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api/aptos_config";
+import { Hex } from "../core";
+import { GetDelegatedStakingActivitiesResponse, GetNumberOfDelegatorsForAllPoolsResponse, HexInput } from "../types";
+import {
+  GetDelegatedStakingActivitiesQuery,
+  GetNumberOfDelegatorsForAllPoolsQuery,
+  GetNumberOfDelegatorsQuery,
+} from "../types/generated/operations";
+import {
+  GetDelegatedStakingActivities,
+  GetNumberOfDelegators,
+  GetNumberOfDelegatorsForAllPools,
+} from "../types/generated/queries";
+import { queryIndexer } from "./general";
+
+export async function getNumberOfDelegators(args: {
+  aptosConfig: AptosConfig;
+  poolAddress: HexInput;
+}): Promise<number> {
+  const { aptosConfig, poolAddress } = args;
+  const address = Hex.fromHexInput({ hexInput: poolAddress }).toString();
+  const query = {
+    query: GetNumberOfDelegators,
+    variables: { poolAddress: address },
+  };
+  const data: GetNumberOfDelegatorsQuery = await queryIndexer<GetNumberOfDelegatorsQuery>({ aptosConfig, query });
+  if (data.num_active_delegator_per_pool.length === 0) {
+    throw Error("Delegator pool not found");
+  }
+  return data.num_active_delegator_per_pool[0].num_active_delegator;
```

Comment:
[this](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/types/indexer.ts#L43) how we handle/format the response. It is to deal with this weird scenario and also to support filtering. You can follow [getAccountOwnedTokens](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/account.ts#L218)